assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,157 @@
+
+console.log(addToCollection('evermore', 'Taylor Swift', 2020));
+console.log(addToCollection('Midnights', 'Taylor Swift', 2022));
+console.log(addToCollection('folklore', 'Taylor Swift', 2020));
+console.log(addToCollection('Endless Summer', 'Miley Cyrus', 2023));
+console.log(addToCollection('Lemonade', 'Beyonce', 2016));
+console.log(addToCollection('Happier Than Ever', 'Billie Eilish', 2021));
+console.log(collection);
+
+// - Add a function named `showCollection`. This function should:
+//   - Take in an array parameter. (This allows it to be reused to show any collection, like the results from the find or search.)
+//   - Console.log the number of items in the array.
+//   - Loop over the array and console.log each album's information formatted like: `TITLE by ARTIST, published in YEAR`.
+
+function showCollection(array) {
+    console.log(array.length);
+    for(let obj of array) {
+        console.log(`${obj.title} by ${obj.artist}, published in ${obj.year}`)
+    }
+}
+
+// - Test the `showCollection` function.
+
+showCollection(collection);
+
+// - Add a function named `findByArtist`. This function should:
+//   - Take in `artist` (a string) parameter
+//   - Create an array to hold any results, empty to start
+//   - Loop through the `collection` and add any objects with a matching artist to the array.
+//   - Return the array with the matching results. If no results are found, return an empty array.
+
+function findByArtist(newArtist) {
+    let artistArray = [];
+    for (let obj of collection) {
+        if(newArtist === obj.artist){
+            artistArray.push(obj);
+        }
+    }
+    return artistArray;
+}
+
+// - Test the `findByArtist` function. Make sure to test with an artist you know is in the collection, as well as an artist you know is not in your collection. Check that for artists with multiple matches, all are found.
+
+console.log(findByArtist('Beyonce'));
+console.log(findByArtist('Taylor Swift'));
+console.log(findByArtist('Selena Gomez'));
+
+// > When testing your functions, write all tests in the JavaScript file!
+
+
+// ### Stretch goals
+
+// - Create a function called `search`. This function should:
+//   - Take an input parameter for a search criteria object. Create your solution based on a search object that has these properties:
+//   ```
+//   { artist: 'Ray Charles', year: 1957 }
+//   ```
+//   - The returned output from `search` should meet these requirements:
+//     - Return a new array of all items in the `collection` matching *all* of the search criteria.
+//     - If no results are found, return an empty array.
+//     - If there is no search object or an empty search object provided as input, then return all albums in the `collection`.
+
+
+let searchObject = {
+    artist: 'Ray Charles',
+    year: 1957
+}
+
+let taylorSwift = {
+    artist: 'Taylor Swift',
+    year: 2020
+}
+
+function search(searchCriteriaObject) {
+    if(arguments.length < 1) {
```

Comment:
Someone has been doing some googling!

This works, and well done, but FYI the `arguments` object is not used much anymore. It has pretty much entirely been replaced by the [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters).

```js
function greetMany(...names) {
  console.log('Hello', names);
};

greetMany();                // Hello []
greetMany('Anne');          // Hello ['Anne']
greetMany('Bob', 'Steve');  // Hello ['Bob', 'Steve']
```

That said, for _this_ case, a more common approach would be to just check if `searchCriteriaObject` is `undefined` or not:

```suggestion
    if(!searchCriteriaObject) {
```